PublicMapping/districtbuilder
# Choose fields to use when calculating demographics

## Overview

Certain custom regions currently have data for both VAP (voting age population) as well as CVAP (citizen voting age population), in addition to our normal demographics data. Additionally, we plan to add VAP data for our existing active 50 state regions.

To make that data more useful, we'd like to enable using those fields when displaying demographic information & when calculating majority race & majority-minority district status, which this PR adds support for in the form of a new option in the map settings menu.

### Checklist


### Demo

![Animation (1)](https://user-images.githubusercontent.com/4432106/146394761-1e63a5b5-b64d-466a-98f6-ce61ca11a7d3.gif)



## Testing Instructions

- Create a map using a region with multiple demographic groups, like the new San Bernadino region added to `load-dev-data` which has VAP & CVAP groups.
  - You should see a new option in the ⚙️ flyout menu allowing you to change the population field. Changing it should be reflected in the:
    - Sidebar charts and tooltips
    - Sidebar / Evaluate Majority race calculation
    - Sidebar / Evaluate Majority-Minority district status
    - Map tooltip chart
- The option to change the population field should not be present in the ⚙️ menu for regions without more than 1 demographic group in their `static-metadata.json`, like the other existing test regions.

Closes #1042
